### DISCUSSION TITLE
How to calculate agent_obs_shape and global_obs_shape in configuration file of smac env?

### DISCUSSION THREAD

**Discussion by Monarchlee** [opendilab/DI-engine#584] [Q&A] [answered]
I am confused by the calculation of agent_obs_shape and global_obs_shape. The docs just give me the value without any process about these two variables but I want to train agents on a self-made map. The official code is below.

```policy=dict(
        cuda=True,
        multi_agent=True,
        continuous=False,
        model=dict(
            # (int) agent_num: The number of the agent.
            # For SMAC 3s5z, agent_num=8; for 2c_vs_64zg, agent_num=2.
            agent_num=agent_num,
            # (int) obs_shape: The shapeension of observation of each agent.
            # For 3s5z, obs_shape=150; for 2c_vs_64zg, agent_num=404.
            # (int) global_obs_shape: The shapeension of global observation.
            # For 3s5z, obs_shape=216; for 2c_vs_64zg, agent_num=342.
            agent_obs_shape=150,
            #global_obs_shape=216,
            global_obs_shape=295,
            # (int) action_shape: The number of action which each agent can take.
            # action_shape= the number of common action (6) + the number of enemies.
            # For 3s5z, obs_shape=14 (6+8); for 2c_vs_64zg, agent_num=70 (6+64).
            action_shape=14,
            # (List[int]) The size of hidden layer
            # hidden_size_list=[64],
        )
```

Could anyone tell me how to calculate agent_obs_shape and global_obs_shape? Also, is it related to the type of entity or the kind of algorithm?

**Comment by yifan123** [comment, marked as answer]
1.  It is related to the type of entity
2. agent_obs https://github.com/opendilab/DI-engine/blob/f1f0b55b42a813dc6d82f0395db4ef495026f752/dizoo/smac/envs/smac_env.py#L1041
3. global_obs shape https://github.com/opendilab/DI-engine/blob/f1f0b55b42a813dc6d82f0395db4ef495026f752/dizoo/smac/envs/smac_env.py#L1198
4. You can print the shape in [smac env](https://github.com/opendilab/DI-engine/blob/f1f0b55b42a813dc6d82f0395db4ef495026f752/dizoo/smac/envs/smac_env.py#L591) and then set the right value in config and run again

**Reply by Monarchlee** [reply]
Thanks a lot. I will try to print the shape then.
